src/main/java/com/google/ehub/servlets/FavoriteCounterServlet.java

```diff
@@ -0,0 +1,37 @@
+package com.google.ehub.servlets;
+
+import com.google.ehub.data.FavoriteItemDatastore;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.lang3.math.NumberUtils;
+
+/**
+ * Handles Get requests to get the users that liked a specific entertainment item.
+ */
+@WebServlet("/favorite-counter")
+public class FavoriteCounterServlet extends HttpServlet {
+  private static final String ITEM_ID_PARAMETER_KEY = "itemId";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String itemId = request.getParameter(ITEM_ID_PARAMETER_KEY);
+
+    if (!isGetRequestParameterValid(itemId)) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST,
+          "FavoriteCounterServlet: Get request parameter was not specified correctly!");
+      return;
+    }
+
+    response.setContentType("application/json");
+    response.getWriter().println(
```

Comment:
Do you really need to expose all the actual emails? What will the client do with all the emails instead of just the count?

Reply:
I thought about keeping the functionality more generic if for some reason I wanted to know the emails for other features. I could remove it and just send the size.

Reply:
It's good practice to be very cautious with personal information and only expose it if absolutely necessary

Reply:
OK I am going to only expose the size of the list.

Reply:
OK I changed it so that it only sends the size of the list in last commit.